From 22a177179cf37f24ef9e24e31801a3a2bb8a99b3 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Wed, 23 Oct 2024 08:23:59 +0300
Subject: [PATCH] [SW-205224] gaudi3: add d2d spi clock init

This patch adds support for fine tuning of d2d
spi clock as per system team recommendation

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ic8d0c98a7539057d9ea6376e61b6f25135ea2b0c
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index a90ae90..afb110a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -365,6 +365,13 @@
 	/* psoc drain is enabled in HW by default */
 }
 
+#define D2D_1V2_CFG_VAL			0x10
+static void d2d_spi_clk_init(void)
+{
+	regmap_writel(D2D_1V2_CFG_VAL, D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_1V2_CFG_34));
+
+}
+
 void hl_plat_early_init(void)
 {
 	hl_psoc_drain_init();
@@ -372,6 +379,7 @@
 	init_prstn();
 	hl_dev_cpu_pll_init();
 	hl_init_cpu_ids();
+	d2d_spi_clk_init();
 }
 
 #define AGG_SEI_PARC_INT_IDX_STS			BIT(25)
